### PR TITLE
Implement overridable process args

### PR DESCRIPTION
**Pull request by natalieparellano**
Thanks to the awesome work of @jabrown85 in https://github.com/buildpacks/lifecycle/pull/920 and https://github.com/buildpacks/lifecycle/pull/889, this PR is relatively small. I'll add comments to show user acceptance.

### CODE REVIEW

File: launch/launch_test.go [hunk 1]
```diff
@@ -18,29 +19,97 @@ func TestLaunch(t *testing.T) {
...
 	when("Process", func() {
 		when("MarshalText", func() {
-			it("command is string", func() {
+			it("command is array", func() {
 				process := launch.Process{
 					Type:             "some-type",
-					Command:          []string{"some-command"},
-					Args:             []string{"some-arg"},
+					Command:          []string{"some-command", "some-arg1"},
+					Args:             []string{"some-arg2"},
```
Comment (jabrown85) [inline launch/launch_test.go:26]:
```suggestion
					Args:             []string{"some-arg1"},
```

**Comment by natalieparellano** [comment]
Summary of changes for posterity:

### HAVE

#### Build

| Buildpack writes launch.toml   | Metadata struct           | Lifecycle writes metadata file |
|--------------------------------|---------------------------|--------------------------------|
| (Old bp)                       |                           |                                |
| Command = "string"             | Command = ["string"]      | Command = "string"             |
| Args = ["arg1", "arg2"]        | Args = ["arg1", "arg2"]   | Args = ["arg1", "arg2"]        |
| ------------------------------ | ------------------------- | -------------------------      |
| (New bp)                       |                           |                                |
| Command = ["string", "arg1"]   | Command = ["string"]      | Command = "string"             |
| Args = ["arg2"]                | Args = ["arg1", "arg2"]   | Args = ["arg1", "arg2"]        |

#### Run

| Metadata file                            | Metadata struct                          | Launcher                                     |
|------------------------------------------|------------------------------------------|----------------------------------------------|
| Command = "string"                       | Command = ["string"]                     | launches "string" with args "arg1", "arg2"   |
| Args = ["arg1", "arg2"]                  | Args = ["arg1", "arg2"]                  |
| -------------------------                | -------------------------                | -------------------------------------------- |
| Command = ["string", "would-not-happen"] | Command = ["string", "would-not-happen"] | launches "string" with args "arg1", "arg2"   |
| Args = ["arg1", "arg2"]                  | Args = ["arg1", "arg2"]                  |                                              |


### WANT

#### Build

| Buildpack writes launch.toml   | Metadata struct              |
|--------------------------------|------------------------------|
| (Old bp)                       |                              |
| Command = "string"             | Command = ["string"]         |
| Args = ["arg1", "arg2"]        | Args = ["arg1", "arg2"]      |
| ------------------------------ | -------------------------    |
| (New bp)                       |                              |
| Command = ["string", "arg1"]   | Command = ["string", "arg1"] | <- need to preserve arg1 in Command
| Args = ["arg2"]                | Args = ["arg2"]              |


| Metadata struct                  | Lifecycle writes metadata file |
|----------------------------------|--------------------------------|
|                                  | (Old platform)                 |
| Command = ["string", "arg1"]     | Command = "string"             |
| Args = ["arg2"]                  | Args = ["arg1", "arg2"]        |
| -------------------------------- | ------------------------------ |
|                                  | (New platform)                 |
| Command = ["string", "arg1"]     | Command = ["string", "arg1"]   | <- needs to conditionally happen based on platform API
| Args = ["arg2"]                  | Args = ["arg2"]                |

#### Run

| Metadata file                   | Metadata struct                 | Launcher                                          |
|---------------------------------|---------------------------------|---------------------------------------------------|
| Command = "string"              | Command = ["string"]            | launches "string" with args "arg1", "arg2"        |
| Args = ["arg1", "arg2"]         | Args = ["arg1", "arg2"]         |
| -------------------------       | -------------------------       | --------------------------------------------      |
| Command = ["string", "cmd-arg"] | Command = ["string", "cmd-arg"] | launches "string" with args "cmd-arg", "user-arg" | <- need to check if Command has Len > 1
| Args = ["arg1", "arg2"]         | Args = ["arg1", "arg2"]         |                                                   |

**Review by jabrown85 — APPROVED** [review]
The recent changes look great

**Comment by natalieparellano** [comment]
User acceptance: https://gist.github.com/natalieparellano/727d3e5bba14160ea2a7dad88ccbcfbe

**Pull request merged**